src/Physics/Resonance/XSection/BostedChristyEMPXSec.cxx

```diff
@@ -0,0 +1,402 @@
+  
+  // Get kinematical parameters
+  const Kinematics & kinematics = interaction -> Kine();
+  const InitialState & init_state = interaction -> InitState();
+  
+  double E  = init_state.ProbeE(kRfHitNucRest);
+  double W  = kinematics.W();
+  double Q2 = kinematics.Q2();
+  double W2 = W*W;
+  double Mp = 0.9382727;
+  double Mp2 = Mp*Mp;
+  
+  double sigmaT = sigmaR(0, Q2, W) + sigmaNR(0, Q2, W);
+  double sigmaL = sigmaR(1, Q2, W) + sigmaNR(1, Q2, W);
+
+  double nu = (W2-Mp2+Q2)/2./Mp;
+  double nu2 = nu*nu;
+  double Eprime = E - nu;
+  double sin2theta_2 = Q2/4/E/Eprime;
+  double cos2theta_2 = 1 - sin2theta_2;
+  double tan2theta_2 = sin2theta_2/cos2theta_2;
+  double eps = 1./(1. + 2*(1+nu2/Q2)*tan2theta_2);            // Eq. (4)
+  double Gamma = kAem*Eprime*(W2-Mp2)/Q2/Mp/E/(1-eps)/4/kPi2;   // Eq. (5)
+  double xsec = Gamma*(sigmaT+eps*sigmaL);                // Eq. (3) d2xsec/dOmegadEprime
+  double jacobian = W*kPi/E/Eprime/Mp;  
+  xsec*= jacobian;                                            // d2xsec/dOmegadEprime-> d2xsec/dWdQ2
+  
+  // The algorithm computes d^2xsec/dWdQ2
+  // Check whether variable tranformation is needed
+  if(kps!=kPSWQ2fE) {
+    double J = utils::kinematics::Jacobian(interaction,kPSWQ2fE,kps);
+    xsec *= J;
+  }
+  
+  return xsec*units::ub;                              
+  
+}
+//____________________________________________________________________________
+double BostedChristyEMPXSec::Integral(const Interaction * interaction) const
+{
+  double xsec = fXSecIntegrator->Integrate(this,interaction);
+  return xsec;
+}
+//____________________________________________________________________________
+bool BostedChristyEMPXSec::ValidProcess(const Interaction * interaction) const
+{
+  if(interaction->TestBit(kISkipProcessChk)) return true;
+
+  const InitialState & init_state = interaction->InitState();
+  const ProcessInfo &  proc_info  = interaction->ProcInfo();
+
+  if(!proc_info.IsDeepInelastic()) return false;
+ 
+
+  int  hitnuc = init_state.Tgt().HitNucPdg();
+  bool is_pn = (pdg::IsProton(hitnuc) || pdg::IsNeutron(hitnuc));
+
+  if (!is_pn) return false;
+
+  int  probe   = init_state.ProbePdg();
+  bool is_em   = proc_info.IsEM();
+
+  bool l_em    = (pdg::IsChargedLepton(probe) && is_em  );
+
+  if (!l_em) return false;
+
+  return true;
+}
+
+//____________________________________________________________________________
+void BostedChristyEMPXSec::Configure(const Registry & config)
+{
+  Algorithm::Configure(config);
+  this->LoadConfig();
+}
+//____________________________________________________________________________
+void BostedChristyEMPXSec::Configure(string config)
+{
+  Algorithm::Configure(config);
+  this->LoadConfig();
+}
+//____________________________________________________________________________
+void BostedChristyEMPXSec::LoadConfig(void)
+{  
+  // single pion branching ratios
+  fbr[0][0] = 1.0;       //  P33(1232)       
```

Comment:
All these parameters should be defined in the xml file and loaded here.

Reply:
I can't agree. All these are parameters of their fit, that will never be changed (otherwise it will not be Boosted-Christ fit), so I hardcoded them as in case of resonance masses and widths.